### PR TITLE
converter: designate maps, and supply mapKey

### PR DESCRIPTION
**Pull request by zack-littke-smith-ai**
Schema change should be visible in the generated files

### CODE REVIEW

File: internal/converter/types.go [hunk 1]
```diff
@@ -618,7 +630,7 @@ func (c *Converter) recursiveConvertMessageType(curPkg *ProtoPackage, msgDesc *d
 			c.logger.WithError(err).WithField("field_name", fieldDesc.GetName()).WithField("message_name", msgDesc.GetName()).Error("Failed to convert field")
 			return nil, err
 		}
-		c.logger.WithField("field_name", fieldDesc.GetName()).WithField("type", recursedJSONSchemaType.Type).Trace("Converted field")
+		c.logger.WithField("field_name", fieldDesc.GetName()).WithField("type", recursedJSONSchemaType.Type).WithField("mapKey", recursedJSONSchemaType.MapKey).Trace("Converted field")
```
Comment (zack-littke-smith-ai) [inline internal/converter/types.go:633]: Probably wasn't necessary

File: internal/converter/types.go [hunk 2]
```diff
@@ -342,6 +342,18 @@ func (c *Converter) convertField(curPkg *ProtoPackage, desc *descriptor.FieldDes
 			}
 			jsonSchemaType.AdditionalProperties = additionalPropertiesJSON
 
+			switch *recordType.Field[0].Type {
```
Comment (zack-littke-smith-ai) [inline internal/converter/types.go:345]: Field[0] for maps is the key field

File: internal/converter/testdata/maps.go [hunk 3]
```diff
@@ -6,19 +6,44 @@ const Maps = `{
...
+                        "$ref": "#/definitions/samples.PayloadMessage",
+                        "additionalProperties": true
+                    },
+                    "type": "object"
+                },
+                "map_of_ints_to_strings": {
+                    "mapKey": "integer",
+                    "additionalProperties": {
+                        "type": "string"
+                    },
```
Comment (rami-manna-ai) [inline internal/converter/testdata/maps.go:35]: can we just make this `valueType` and not nest it under `additionalProperties`?

Comment (zack-littke-smith-ai) [reply internal/converter/testdata/maps.go:35]: This seems like a larger change to the schema? I intentionally didnt touch this pathway, seems best to conform as closely as before

Comment (rami-manna-ai) [reply internal/converter/testdata/maps.go:35]: oh nvm, this was already there. disregard! thanks.

**Review by rami-manna-ai — APPROVED** [review]

**Pull request merged**